geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/BinnedArray.scala

```diff
@@ -254,47 +262,45 @@ class BinnedDoubleArray(length: Int, bounds: (jDouble, jDouble)) extends BinnedA
 class BinnedStringArray(length: Int, bounds: (String, String)) extends BinnedArray[String](length, bounds) {
 
   private val (start, end) = {
-    val lower = bounds._1.toLowerCase(Locale.US).replaceAll("[^0-9a-z]", "0")
-    val upper = bounds._2.toLowerCase(Locale.US).replaceAll("[^0-9a-z]", "0")
+    val lower = normalize(bounds._1)
+    val upper = normalize(bounds._2)
     if (lower.length == upper.length) {
       (lower, upper)
     } else if (lower.length < upper.length) {
       (lower.padTo(upper.length, '0'), upper)
     } else {
-      (lower, upper.padTo(upper.length, 'z'))
+      (lower, upper.padTo(lower.length, 'z'))
     }
   }
 
-  require(start < end, s"Upper bound must be greater than lower bound: lower=${bounds._1} upper=${bounds._2}")
-
-  private val firstDiff = start.zip(end).indexWhere { case (l, r) => l != r }
-  private val zeroInt = '0'.toInt
-  private val nineInt = '9'.toInt - zeroInt
-  private val alphaOffset = 'a'.toInt - 10
+  require(start <= end, s"Upper bound must be greater than lower bound: lower=${bounds._1} upper=${bounds._2}")
 
-  def stringToLong(s: String): Long = {
-    val base36 = s.slice(firstDiff, start.length).replaceAll("[^0-9a-z]", "0")
-    if (base36.length > 12) {
-      // won't fit in a long...
-      jLong.parseLong(base36.substring(0, 12), 36)
-    } else {
-      jLong.parseLong(base36, 36)
-    }
+  // # of base 36 numbers we can consider
+  private val sigDigits = math.max(1, math.round(math.log(length) / math.log(36)).toInt)
```

Comment:
Guess I'm not sure - I think that if we used `ceil` we would end up with a lot of empty buckets. If you have any better ideas for putting strings into buckets I'd appreciate it. I also wanted to handle common prefixes, so that e.g. if all your data starts with 'user_id_xxx' everything doesn't end up in the same bucket.